dictionaries/PropertyMap.php

```diff
@@ -2,193 +2,113 @@
 namespace Psalm\Internal;
 
 /**
- * Stolen from https://github.com/etsy/phan/blob/master/src/Phan/Language/Internal/PropertyMap.php
+ * Automatically created by bin/update-property-map.php
  *
- * The MIT License (MIT)
- * Copyright (c) 2015 Rasmus Lerdorf
- * Copyright (c) 2015 Andrew Morrison
- *
- * Permission is hereby granted, free of charge, to any person obtaining a copy
- * of this software and associated documentation files (the "Software"), to deal
- * in the Software without restriction, including without limitation the rights
- * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
- * copies of the Software, and to permit persons to whom the Software is
- * furnished to do so, subject to the following conditions:
- *
- * The above copyright notice and this permission notice shall be included in all
- * copies or substantial portions of the Software.
- *
- * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
- * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
- * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
- * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
- * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
- * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
- * SOFTWARE.
+ * Please do not modify - adapt the override constants in above file instead.
  */
 
 return [
-    'arrayobject' => [
-        'name' => 'string',
-    ],
-    'limititerator' => [
-        'name' => 'string',
-    ],
-    'solrdocumentfield' => [
-        'name' => 'string',
-        'boost' => 'float',
-        'values' => 'array',
-    ],
-    'domprocessinginstruction' => [
-        'target' => 'string',
-        'data' => 'string',
-    ],
-    'recursivearrayiterator' => [
-        'name' => 'string',
-    ],
-    'eventbuffer' => [
-        'length' => 'int',
-        'contiguous-space' => 'int',
-    ],
-    'mongocursor' => [
-        'slaveokay' => 'boolean',
-        'timeout' => 'integer',
-    ],
-    'domxpath' => [
-        'document' => 'DOMDocument',
-    ],
-    'domentity' => [
-        'publicId' => 'string',
-        'systemId' => 'string',
-        'notationName' => 'string',
-        'actualEncoding' => 'string',
-        'encoding' => 'string',
-        'version' => 'string',
+    'CommonMark\\Node' => [
```

Comment:
I think the offset should be a lowercase string

Reply:
Shall I also lowercase the property names?
They were so in the original file but I currently keep them as documented in the XML files.

Reply:
Interesting:
https://github.com/vimeo/psalm/blob/97e6511fab6622c1ca434c665dd895e77a0eeca2/src/Psalm/Internal/Codebase/PropertyMap.php#L30

Psalm does a strtolower by itself on every class name. It would be best to do it on generation instead of in execution (we could drop the whole loop then). There is no such requirement on property names (I'm also not sure they're case insensitive)

Reply:
using strtolower when using propertymap is expected, we'll retrieve all sort of case from user code that we need to normalize